diff -p -up libmpeg3/mpeg3tocutil.c~ libmpeg3/mpeg3tocutil.c
--- libmpeg3/mpeg3tocutil.c~	2008-06-21 05:53:27.000000000 +0200
+++ libmpeg3/mpeg3tocutil.c	2009-07-09 16:14:04.000000000 +0200
@@ -1278,7 +1278,7 @@ void mpeg3_stop_toc(mpeg3_t *file)
 
 // Store file information
 	PUT_INT32(FILE_INFO);
-	fprintf(file->toc_fd, file->fs->path);
+	fprintf(file->toc_fd, "%s", file->fs->path);
 	for(j = strlen(file->fs->path); j < MPEG3_STRLEN; j++)
 			fputc(0, file->toc_fd);
 	PUT_INT64(file->source_date);
@@ -1309,7 +1309,7 @@ void mpeg3_stop_toc(mpeg3_t *file)
 // Path
 		PUT_INT32(TITLE_PATH);
 
-		fprintf(file->toc_fd, title->fs->path);
+		fprintf(file->toc_fd, "%s", title->fs->path);
 
 // Pad path with 0
 		for(j = strlen(title->fs->path); j < MPEG3_STRLEN; j++)
